ecl/hql/hqlexpr.cpp

```diff
@@ -1070,77 +1070,80 @@ bool HqlParseContext::checkEndMeta()
     return wasGathering;
 }
 
-void HqlParseContext::finishMeta(bool isSeparateFile, bool success, bool generateMeta, bool canCache, bool isMacro, IHqlExpression * simplifiedDefinition)
+void HqlParseContext::createCache(IHqlExpression * simplifiedDefinition, bool isMacro)
 {
-    if (metaStack.empty())  // paranoid - could only happen on an internal error
-        return;
+    assertex(simplifiedDefinition);
```

Comment:
I'm not sure if this is correct (e.g., for macros).  I have added HPCC-19623 to ensure it is revisited later.